lib/srv/sess.go

```diff
@@ -349,18 +349,19 @@ func (s *SessionRegistry) leaveSession(party *party) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	// Emit session leave event to both the Audit Log as well as over the
-	// "x-teleport-event" channel in the SSH connection.
-	s.emitSessionLeaveEvent(party)
-
 	sess.mu.Lock()
 	defer sess.mu.Unlock()
-
 	// Remove member from in-members representation of party.
 	if err := sess.removeParty(party); err != nil {
 		return trace.Wrap(err)
 	}
 
+	sess.mu.Unlock()
```

Comment:
This is suuuuuuper confusing. There must be a better way.

Reply:
Changed to use defer instead